### PR TITLE
add hasclaimed hook

### PR DESCRIPTION
**Pull request by L03TJ3**
# Description

Add hook to check if connected account has any claimable UBI on a supported chain

**About**
https://github.com/GoodDollar/GoodProtocolUI/pull/336

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

# Checklist:
- [ ] PR title matches follow: (Feature|Bug|Chore) Task Name
- [ ] My code follows the style guidelines of this project
- [ ] I have followed all the instructions described in the initial task (check Definitions of Done)
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added reference to a related issue in the repository
- [ ] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request (for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [ ] @mentions of the person or team responsible for reviewing proposed changes

### CODE REVIEW

**Comment by sirpy** [comment]
@L03TJ3 is this required for the new claim flow?

**Comment by L03TJ3** [comment]
@sirpy yes that was the idea. We show that a claim is available on the alternative chain.
I needed something like this to check on both chains if someone has claimed already or not.

You have another suggestion?

**Review by sirpy — CHANGES_REQUESTED** [review]
1. you are pointing to beta sdk packages. once its ready to merge it should work with the non beta package.
2. i'd change it to useHasClaimed(requiredNetwork)
and in your use case call it twice. useHasClaimed(fuse) useHasClaimed(celo).

File: packages/sdk-v2/src/sdk/claim/react.ts [hunk 1]
```diff
@@ -87,6 +87,26 @@ export const useClaim = (refresh: QueryParams["refresh"] = "never") => {
   };
 };
 
+export const useHasClaimed = (requiredNetwork: keyof typeof SupportedV2Networks) => {
```
Comment (johnsmith-gooddollar) [inline packages/sdk-v2/src/sdk/claim/react.ts:90]:
```suggestion
export const useHasClaimed = (requiredNetwork: keyof typeof SupportedV2Networks): boolean => {
```

**Review by johnsmith-gooddollar — CHANGES_REQUESTED** [review]

File: packages/sdk-v2/src/sdk/claim/react.ts [hunk 2]
```diff
@@ -87,6 +87,26 @@ export const useClaim = (refresh: QueryParams["refresh"] = "never") => {
...
+        method: "checkEntitlement(address)",
+        args: [account]
+      }
+    ],
+    { refresh: "never", chainId: SupportedV2Networks[requiredNetwork] as unknown as ChainId }
+  );
+
+  return {
+    hasClaimed: first(hasClaimed?.value) as boolean
+  };
```
[the remark this answers is not in the document]
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/claim/react.ts]: Well, it first returned two.

But you are right, doesn't make sense to keep the object return now

File: packages/sdk-v2/src/sdk/claim/react.ts (hunk 1 above)
Comment (L03TJ3) [reply packages/sdk-v2/src/sdk/claim/react.ts]: done

**Pull request merged**